packages/static-build/src/utils/gatsby.ts

```diff
@@ -35,35 +35,34 @@ export async function injectPlugins(
   }
 
   if (pluginsToInject.length === 0) {
-    return;
+    return false;
   }
 
   await addGatsbyPackage(dir, pluginsToInject);
 
   const gatsbyConfigPathTs = path.join(dir, `${GATSBY_CONFIG_FILE}.ts`);
+  const gatsbyConfigPathMjs = path.join(dir, `${GATSBY_CONFIG_FILE}.mjs`);
+  const gatsbyConfigPathJs = path.join(dir, `${GATSBY_CONFIG_FILE}.js`);
   if (await fileExists(gatsbyConfigPathTs)) {
     printInjectingPlugins(pluginsToInject, gatsbyConfigPathTs);
-    return updateGatsbyTsConfig(gatsbyConfigPathTs, pluginsToInject);
-  }
-
-  const gatsbyConfigPathMjs = path.join(dir, `${GATSBY_CONFIG_FILE}.mjs`);
-  if (await fileExists(gatsbyConfigPathMjs)) {
+    await updateGatsbyTsConfig(gatsbyConfigPathTs, pluginsToInject);
+  } else if (await fileExists(gatsbyConfigPathMjs)) {
     printInjectingPlugins(pluginsToInject, gatsbyConfigPathMjs);
-    return updateGatsbyMjsConfig(gatsbyConfigPathMjs, pluginsToInject);
-  }
-
-  const gatsbyConfigPathJs = path.join(dir, `${GATSBY_CONFIG_FILE}.js`);
-  printInjectingPlugins(pluginsToInject, gatsbyConfigPathJs);
-  if (await fileExists(gatsbyConfigPathJs)) {
-    await updateGatsbyJsConfig(gatsbyConfigPathJs, pluginsToInject);
+    await updateGatsbyMjsConfig(gatsbyConfigPathMjs, pluginsToInject);
   } else {
-    await fs.writeFile(
-      gatsbyConfigPathJs,
-      `module.exports = ${JSON.stringify({
-        plugins: pluginsToInject,
-      })}`
-    );
+    printInjectingPlugins(pluginsToInject, gatsbyConfigPathJs);
+    if (await fileExists(gatsbyConfigPathJs)) {
+      await updateGatsbyJsConfig(gatsbyConfigPathJs, pluginsToInject);
+    } else {
+      await fs.writeFile(
+        gatsbyConfigPathJs,
+        `module.exports = ${JSON.stringify({
```

Comment:
Perhaps we should use `.cjs` extension for `module.exports` (or use the `.mjs` with `export default`

Reply:
We'd have to check if gatsby supports the .cjs extension for its config file

Reply:
And we do use .mjs with export syntax

Reply:
@Ethan-Arrowood What I mean is that `.js` is ambiguous and depends on the customer's `package.json`

I think we should be explicit when writing to a new file.

Reply:
this is a gatsby thing unfortunately https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/